### ISSUE TITLE
We need an IResponse interface

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5057]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#5057|
|Type|enhancement|
|Created|2011-04-19 18:09:40Z|
|Branch|https://github.com/twisted/twisted/tree/iresponse-5057|

To be able to wrap web client Response class properly, an interface would help. It would also have some documentation benefits.

<details><summary>Searchable metadata</summary>

```
trac-id__5057 5057
type__enhancement enhancement
reporter__therve therve
priority__normal normal
milestone__ 
branch__branches_iresponse_5057 branches/iresponse-5057
branch_author__therve therve
status__closed closed
resolution__fixed fixed
component__web web
keywords__easy easy
time__1303236580000000 1303236580000000
changetime__1305233875000000 1305233875000000
version__None None
owner__therve therve
cc__jknight
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

(In [31702]) Branching to 'iresponse-5057'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve removed owner|
|-|-|

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @therve|
|-|-|

  1. `IResponse` should be added to `twisted.web.iweb.__all__`.
  1. If we want `IResponse` to be the main thing people look at (instead of `Response`), maybe all of the docs on `Response` should be on `IResponse` (eg, the rest of the `deliverBody` docstring and the type information for the attributes) and the `Response` docstrings can focus on just the particulars of its implementation?
  1. `IResponse` should have `@since` I guess (we should automate that though)

Otherwise looks good ([build results, too](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/iresponse-5057)), merge when those are fixed.  Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set status to `closed`|
|-|-|

(In [31820]) Merge iresponse-5057

Author: therve
Reviewier: exarkun
Fixes: [#5057](https://github.com/twisted/twisted/issues/5057)

Create a IResponse interface to represent the Response returned by
twisted.web.client.Agent.deliverBody.
